### PR TITLE
`1ES-Template` Conversion

### PR DESCRIPTION
**Pull request by scbedd**
[Successful Template Release](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3620385&view=logs&j=b0bfa1f3-db9a-5134-fed5-79a504466d6e)

Merging after successful template release.

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -25,34 +27,38 @@ jobs:
...
                      -configuration Release \
-                     -archivePath $(Build.ArtifactStagingDirectory)/AzureSDK-iphonesimulator.xcarchive \
+                     -archivePath $(Build.ArtifactStagingDirectory)/iphonesimulator/AzureSDK-iphonesimulator.xcarchive \
                      SKIP_INSTALL=NO | xcpretty -c
         displayName: 'Build and archive libraries for release [iphonesimulator]'
 
-      - publish: $(Build.ArtifactStagingDirectory)/AzureSDK-iphonesimulator.xcarchive
-        displayName: 'Publish release archive [iphonesimulator]'
-        artifact: 'archive-release-iphonesimulator'
+      - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
```
Comment (benbp) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:41]:
```suggestion
      - template: /eng/common/pipelines/templates/steps/publish-1es-artifact.yml
```

Comment (benbp) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:41]: Or is there a specific reason for the old one?

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml [hunk 2]
```diff
@@ -70,9 +71,11 @@ stages:
       - stage: ${{ cloud.key }}
         dependsOn: []
         jobs:
-        - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
+        - template: /eng/common/pipelines/templates/jobs/generate-job-matrix.yml
           parameters:
             JobTemplatePath: /eng/pipelines/templates/jobs/live.tests.yml
+            OsVmImage: $(LINUXVMIMAGE)
+            Pool: $(LINUXPOOL)
```
Comment (benbp) [inline eng/pipelines/templates/stages/archetype-sdk-tests.yml:78]:
```suggestion
```

Not needed

File: eng/pipelines/templates/variables/image.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    value: azsdk-pool-mms-ubuntu-2004-1es-pt
+  - name: WINDOWSPOOL
+    value: azsdk-pool-mms-win-2022-1es-pt
+  - name: MACPOOL
+    value: Azure Pipelines
+
+  - name: LINUXVMIMAGE
+    value: azsdk-pool-mms-ubuntu-2004-1espt
+  - name: LINUXNEXTVMIMAGE
+    value: ubuntu-22.04
```
Comment (benbp) [inline eng/pipelines/templates/variables/image.yml:14]:
```suggestion
  - name: LINUXNEXTVMIMAGE
    value: azsdk-pool-mms-ubuntu-2204-1espt
```

**Review by benbp — APPROVED** [review]
LGTM with a couple nits and a question about the publish task [EDIT] Looks like you beat me to it.

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml (hunk 2 above)
Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml]: **deep sigh**

File: eng/pipelines/templates/variables/image.yml (hunk 3 above)
Comment (scbedd) [reply eng/pipelines/templates/variables/image.yml:14]: ROFL. YOU GOT ME

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml (hunk 1 above)
Comment (scbedd) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml]: It was missed I have since caught it.

**Comment by scbedd** [comment]
Template released successfully!

**Pull request merged**